addons/website_form_project/controllers/main.py

```diff
@@ -2,14 +2,13 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo.addons.website.controllers import form
+from odoo.http import request
 
 
 class WebsiteForm(form.WebsiteForm):
-    def insert_record(self, request, model, values, custom, meta=None):
-        if model.model == 'project.task':
-            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
-            visitor_partner = visitor_sudo.partner_id
-            if visitor_partner:
-                values['partner_id'] = visitor_partner.id
-
-        return super().insert_record(request, model, values, custom, meta=meta)
+    def extract_data(self, model, values):
```

Comment:
Is there a reason we changed where we are doing this operation ? From ``insert_record`` to ``extract_data``

Reply:
The method 'insert_record' serves the purpose of setting the partner ID within the values dictionary. On the other hand, the method 'extract_data' is responsible for checking and directly assigning the partner_id in record dictionary based on the user input received in the form. This approach eliminates the need to set the partner_id separately in the values dictionary during task creation, as it is automatically extracted and added during the process.